src/components/icon/Icon.tsx

```diff
@@ -2,11 +2,12 @@ import React, { ReactNode } from 'react';
 import Box from '../box';
 import cx from 'classnames';
 import theme from './theme.css';
+import { BoxProps } from '../box/Box';
 
 export type COLOURS = 'neutral' | 'mint' | 'violet' | 'ruby' | 'gold' | 'aqua' | 'teal';
 export type TINTS = 'lightest' | 'light' | 'normal' | 'dark' | 'darkest';
 
-interface IconProps {
+interface IconProps extends Omit<BoxProps, 'ref'> {
```

Comment:
I would say omit types should be `children | className` since you are overriding them